src/AbstractServiceProvider.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Dhii\Di;
+
+/**
+ * Basic and common functionality for a service provider.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractServiceProvider
+{
+    /**
+     * The prefix for service IDs provided by this service provider.
+     *
+     * To be overridden by extending classes.
+     *
+     * @since [*next-version*]
+     */
+    const SERVICE_PREFIX = '';
+
+    /**
+     * Parameter-less constructor.
+     *
+     * Call this in the real constructor.
+     *
+     * @since[*next-version*]
+     */
+    protected function _construct()
+    {
+    }
+
+    /**
+     * Retrieves the prefix for service IDs.
+     *
+     * @since [*next-version*]
+     *
+     * @return string
+     */
+    protected function _getServicePrefix()
+    {
+        return static::SERVICE_PREFIX;
+    }
+
+    /**
+     * Prefixes a service ID.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $serviceId The service ID.
+     *
+     * @return string The prefixed service ID.
+     */
+    protected function _p($serviceId)
+    {
+        return sprintf('%1$s%2$s', $this->_getServicePrefix(), $serviceId);
+    }
+
+    /**
+     * A shorthand utility method for retrieving a callback to a method of this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $methodName The name of the method.
+     *
+     * @return callable The callback for the method.
+     */
+    protected function _m($methodName)
+    {
+        return array($this, $methodName);
```

Comment:
An even better idea (and we need to remember this for later) is to use the different, main, branch of the CS Fixer standard package.